### PR TITLE
Version 2.2.0

### PR DESCRIPTION
**Pull request by manuelm**
mostly code cleanup and some minor fixes. No fancy new features.

### CODE REVIEW

File: pvr.dvbviewer/changelog.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
+2.2.0
+[updated] Require DVB Viewer Recording Service 1.30.1 or later
+[updated] implemented PVR_RECORDING.iChannelUid (PVR API v5.0.0)
+[updated] code cleanup
+[fixed] timeshift buffer path: allow escaping from special://userdata
+
 2.1.0
```
Comment (ksooo) [inline pvr.dvbviewer/changelog.txt]: 2.1.1 was released without updating the changelog. Only change was pvr api bump to 5.1.1, no code/other changes. Maybe you want to insert a changelog entry for 2.1.1 anyway?

File: src/DvbData.cpp [hunk 2]
```diff
@@ -488,8 +488,15 @@ bool Dvb::GetRecordings(ADDON_HANDLE handle)
...
-    recinfo.iChannelUid   = PVR_CHANNEL_INVALID_UID; // TODO: try searching by name
-    recinfo.channelType   = PVR_RECORDING_CHANNEL_TYPE_TV;
+    recinfo.iChannelUid   = PVR_CHANNEL_INVALID_UID;
+    recinfo.channelType   = PVR_RECORDING_CHANNEL_TYPE_UNKNOWN;
+
+    if (recording.channel)
+    {
+      recinfo.iChannelUid = recording.channel->id;
+      recinfo.channelType = (recording.channel->radio)
+          ? PVR_RECORDING_CHANNEL_TYPE_RADIO : PVR_RECORDING_CHANNEL_TYPE_TV;
```
Comment (ksooo) [inline src/DvbData.cpp]: just for your info, because the commit msg only mentions api 5.0.0: iChannelUid was api 5.0.0, channelType was api 5.1.0

**Comment by ksooo** [comment]
nice. api changes looking good.

File: src/DvbData.cpp (hunk 2 above)
Comment (manuelm) [inline src/DvbData.cpp]: thanks for the info

File: pvr.dvbviewer/changelog.txt (hunk 1 above)
Comment (manuelm) [inline pvr.dvbviewer/changelog.txt]: I've incorporated the change into 2.2.0 changelog

**Comment by manuelm** [comment]
Any idea about the travis build error? Doesn't look related to my changes.

**Comment by ksooo** [comment]
@FernetMenta  just fixed it. https://github.com/kodi-pvr/pvr.dvbviewer/commit/7a86cc5389d1a1d5b317dd72de58882aa2bd8fc6

**Pull request merged**